### PR TITLE
Simplified lives defines

### LINKED ISSUE #433: Instead of a DISABLE_LIVES define and a DEFAULT_NUM_LIVES define, combine both into a single one.
The define could be like `ENABLE_LIVES X` where `X` is the amount of lives you start with and if you disable the define entirely then lives are disabled.

I'll handle this

### LINKED ISSUE #438: Remove `SAVE_NUM_LIVES` define
let'`s be real

no one is going to use it

and look at this: 
imagen

### PR DESCRIPTION
**Pull request by Reonu**
`ENABLE_LIVES` and `DEFAULT_NUM_LIVES` are now the same define. 

Also, `SAVE_NUM_LIVES` has been removed since it was a super chonky define that, realistically, no one is going to use. 

Fixes #433 
Fixes #438

### CODE REVIEW

**Review by thecozies — COMMENTED** [review]
looks great but have a comment about comments

File: src/game/hud.c [hunk 1]
```diff
@@ -32,7 +32,7 @@
 #define HUD_POWER_METER_HIDDEN_Y     300
 
 #ifdef BREATH_METER
-// #ifdef DISABLE_LIVES
+// #ifndef ENABLE_LIVES
```
[the remark this answers is not in the document]
Comment (thecozies) [reply src/game/hud.c:35]: can you clean that up while you're in here 😔

**Comment by Arceveti** [comment]
If lives are enabled they should be saved by default

**Comment by arthurtilly** [comment]
looks good

**Comment by thecozies** [comment]
needs conflicts resolved @Reonu

**Review by gheskett — APPROVED** [review]
Awaiting another approval before merging since I ended up contributing to this PR

**Review by arthurtilly — APPROVED** [review]
approved

**Pull request merged**